app.js

```diff
@@ -0,0 +1,180 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const express = require('express');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const errorHandler = require('errorhandler');
+const session = require('express-session');
+const passport = require('passport');
+const routes = require('./lib/routes');
+const debug = require('debug')('guestlister');
+const config = require('./config/config');
+
+const app = express();
+const mongojs = require('mongojs');
+const mongoose = require('mongoose');
+const backoff = require('backoff');
+const MongoStore = require('connect-mongodb-session')(session);
+
+// use ES6 promises for mongoose
+mongoose.Promise = global.Promise;
+
+const dbURI = config.mongo.location + config.mongo.database;
+mongoose.connect(dbURI, {
+    promiseLibrary: global.Promise // use ES6 promises for underlying MongoDB DRIVE
+});
+mongoose.connection.on('error', (err) => {
+    debug('Could not connect to MongoDB @ %s: %s', dbURI, err);
+});
+// If the Node process ends, close the Mongoose connection
+process.on('SIGINT', function () {
+    mongoose.connection.close(function () {
+        debug('Mongoose default connection disconnected through app termination signal (SIGINT)');
+        process.exit(0);
+    });
+});
+
+// Passport configuration
+require('./lib/auth');
+
+function initApp(callback) {
+    debug('Initialize application');
+
+    // Express configuration
+    app.set('view engine', 'ejs');
+    app.use(express.static(__dirname + '/css'));
+    app.use(cookieParser());
+    app.use(bodyParser.json({ extended: false }));
+    app.use(bodyParser.urlencoded({ extended: false }));
+    app.use(errorHandler());
+    // todo check if guestlister has to use same session as bouncer
+    app.use(session({ secret: config.sessionsecret, resave: false, saveUninitialized: false }));
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    // Routes
+    app.get('/', routes.site.index);
+    app.get('/login', routes.site.loginForm);
+    app.post('/login', routes.site.login);
+    app.get('/logout', routes.site.logout);
+
+    app.get(config.oauth.authorizationPath, routes.oauth2.authorization);
+    app.post(config.oauth.tokenPath, routes.oauth2.token);
+
+    //Create test users in database
+    debug('Creating test users: %O', config.testUsers);
+
+    const db = mongojs('localhost/' + config.mongo.database, ['users', 'sessions']);
+
+    function saveUserAsync(user) {
+        return new Promise(function(resolve, reject) {
+            db.users.save(user, function (err, doc) {
+                if (err) {
+                    return reject(err);
+                }
+                return resolve([doc]);
+            });
+        });        }
+
+    const authoruser = {
```

Comment:
These `const` variables are 80% taken from config variables.
Please add the `_id` to `config.testUsers` and iterate over the `config.testUsers` array.
You can create an array of function calls and pass that to `Promise.all(..)`.
`lastSeenAt` is not really relevant, can also be fixed in `config`.

Reply:
Maybe just leave out `_id`? It should only matter if we're also inserting sessions, which we are not.